### ISSUE TITLE
Binding dynamic value to style property throw error on iOS. [Doc request]

### ISSUE THREAD

**Issue by lricoy** [angular/angular#8500]
I opened the following issue on the Ionic repo and we got to the conclusion that a documentation mention to [style] being readonly would be in order.
https://github.com/driftyco/ionic/issues/6447#issuecomment-217255985

The docs never mention that style is readonly, although the explanation of how style properties should be used is quite clear, without reading it, it may cause confusion since using style="foo" is quite a common practice.

> Style binding syntax resembles property binding. Instead of an element property between brackets, we start with the prefix style, followed by a dot (.) and the name of a CSS style property: [style.style-property].

**Steps to reproduce and a minimal demo of the problem**
1. Bind a dynamic value to a style property (style="background-color: {{user.color}};)
2. Run on iOS
3. View the debug log

```
ERROR:    <div class="user-avatar" style="background-color: {{user.color}};">
WORKS:    <div class="user-avatar" [ngStyle]="{'background-color': user.color}">
```

**Current behavior**
 It works fine on android and the browser but it breaks when running on WebView

**Expected/desired behavior**
No error.

**Other information**

**Comment by pkozlowski-opensource** [comment]
@lricoy I'm not sure what would be the best place to mention it in the docs, do you have a suggestion? 

The part of the documentation you are bringing up refers to the `[style.foo]="exp"` syntax and AFAIK it works correctly across all supported browsers.

`[style]="exp"` is binding directly to the `style` property of an underlying DOM element and, you have noticed, there impl differences across browsers. But such differences exists for some other properties as well, so documentation shouldn't be `style` specific.

**Comment by lricoy** [comment]
Hey @pkozlowski-opensource thanks for the explanation. at first I thought that if we were able to mark parts of the documentation as `@const` (as the JS `const`)  or `@not-reasignable` it would provide a more automated way to inform that behavior.

Considering your comments, I guess that another option would be to put a warning on the [property binding](https://angular.io/docs/ts/latest/guide/template-syntax.html#!#property-binding) (maybe near side-effects ?) or some other part of the template syntax.

I opened the issue mainly because I think we could mention that behavior somewhere, so it would appear in searches if anyone ever happen to use it like that and get a error, like I did. But I am not so sure where would the best place be.
